features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/ApplicationSettings/EditApplication.jsx

```diff
@@ -0,0 +1,202 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useReducer } from 'react';
+import API from 'AppData/api';
+import PropTypes from 'prop-types';
+import TextField from '@material-ui/core/TextField';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import { FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import FormDialogBase from 'AppComponents/AdminPages/Addons/FormDialogBase';
+import Alert from 'AppComponents/Shared/Alert';
+
+const useStyles = makeStyles((theme) => ({
+    error: {
+        color: theme.palette.error.dark,
+    },
+}));
+
+let initialState = {
+    name: '',
+    owner: '',
+};
+
+/**
+ * Reducer
+ * @param {JSON} state State
+ * @returns {Promise}.
+ */
+function reducer(state, { field, value }) {
+    return {
+        ...state,
+        [field]: value,
+    };
+}
+
+/**
+ * Render a pop-up dialog to change ownership of an Application
+ * @param {JSON} props props passed from parent
+ * @returns {JSX}.
+ */
+function Edit(props) {
+    const classes = useStyles();
+    const {
+        updateList, dataRow, icon, triggerButtonText, title, applicationList,
+    } = props;
+    let id = null;
+
+    if (dataRow) {
+        const { name: originalName, owner: originalOwner } = dataRow;
+        id = dataRow.applicationId;
+
+        initialState = {
+            name: originalName,
+            owner: originalOwner,
+        };
+    }
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const { name, owner } = state;
+
+    const onChange = (e) => {
+        dispatch({ field: e.target.name, value: e.target.value });
+    };
+
+    const validateOwner = () => {
+        const valid = { invalid: false, error: '' };
+
+        const applicationsWithSameName = applicationList.filter(
+            (app) => app.name === name && app.owner === owner,
+        );
+        if (applicationsWithSameName.length > 0) {
+            valid.error = `${owner} already has an application with name: ${name}`;
+            valid.invalid = true;
+        }
+        // todo: Validate whether the owner is an existing subscriber or not.
+
+        return valid;
+    };
+
+
+    const getAllFormErrors = () => {
+        let errorText = '';
+        const valid = validateOwner(applicationList);
+        if (valid.invalid) {
+            errorText += valid.error;
+        }
+
+        return errorText;
+    };
+    const formSaveCallback = () => {
+        const formErrors = getAllFormErrors();
+        if (formErrors !== '') {
+            Alert.error(formErrors);
+            return false;
+        }
+        const restApi = new API();
+        let promiseAPICall = restApi.updateApplicationOwner(id, owner);
+
+        promiseAPICall = new Promise((resolve, reject) => {
```

Comment:
Do we need to create new promise here? 
We can do restApi.updateApplicationOwner(id, owner).then() directly, since it's a promise itself

Reply:
Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8492/commits/707ae1e741f6b5ecddc2c413d0e2635a892c44db

Reply:
Please note that creating `new Promise` is necessary here, since the original promise we get from the `restApi.updateApplicationOwner()` is not giving the resolve to be compatible with the 'ListBase' component, which is supposed to be universal to the admin portal.

This is same for creating new promise in `ListApplications` component as well.